src/main/java/org/iot/dsa/dslink/java/v2/jdbc/JDBCClosedTable.java

```diff
@@ -0,0 +1,227 @@
+package org.iot.dsa.dslink.java.v2.jdbc;
+
+import org.iot.dsa.node.*;
+import org.iot.dsa.node.action.ActionSpec;
+import org.iot.dsa.node.action.ActionTable;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ * Class designed to handle tables for non-streaming queries.
+ *
+ * @author James (Juris) Puchin
+ * Created on 10/13/2017
+ */
+public class JDBCClosedTable implements ActionTable {
+
+    private ActionSpec act;
+    private ResultSet res;
+    private Logger log;
+    private List<DSMap> cols;
+    private int columnCount;
+    private ColType[] colTypes;
+
+    JDBCClosedTable(ActionSpec act, ResultSet res, Logger log) throws SQLException {
+        this.act = act;
+        this.res = res;
+        this.log = log;
+
+        ResultSetMetaData meta = res.getMetaData();
+        columnCount = meta.getColumnCount();
+
+        colTypes = new ColType[columnCount + 1];
+        cols = new LinkedList<>();
+        for (int i = 1; i <= columnCount; i++) {
+            DSValueType type = setColumnType(meta, i);
+            cols.add(makeColumn(meta.getColumnName(i), type));
+        }
+    }
+
+    private static DSMap makeColumn(String name, DSValueType type) {
+        return new DSMetadata().setName(name).setType(type).getMap();
+    }
+
+    private DSValueType setColumnType(ResultSetMetaData meta, int idx) {
+        DSValueType ret = DSValueType.STRING;
+        colTypes[idx] = ColType.TYPE_STRING;
+
+        try {
+            switch (meta.getColumnType(idx)) {
+                //null
```

Comment:
not a fan of those comments. they don't give me super useful info

Reply:
Just a way to visually break up the very long switch statement.